valohai_yaml/schema/endpoint.yaml

```diff
@@ -41,6 +41,44 @@ properties:
       Specifies the WSGI application to serve (e.g. a Flask application).
       Specify the module (i.e. `package.app`) or
       the module and the WSGI callable (i.e. `package.app:wsgi_callable`).
+  resources:
+    type: object
+    description: Resource requirements and limits for the endpoint.
+    additionalProperties: false
+    properties:
+      cpu:
+        type: object
+        description: CPU resource requirements and limits in vCPU counts e.g. 0.1 is 10% of one CPU.
+        additionalProperties: false
+        properties:
+          min:
+            type: number
+          max:
+            type: number
+        anyOf:
+          - required:
+              - min
+          - required:
+              - max
+      memory:
+        type: object
+        description: Memory requirements and limits in MB e.g. 100 is 100 MB.
+        additionalProperties: false
+        properties:
+          min:
+            type: number
+          max:
+            type: number
+        anyOf:
+          - required:
+              - min
+          - required:
+              - max
+    anyOf:
+      - required:
+          - cpu
+      - required:
+          - memory
```

Comment:
What does this specification mean? 🤔 

If it is what I think it is, I don't think we'll need to require either of these to be set if the `resources:` block is present...

Reply:
yeah, my thinking was that if `resources` object is defined, it must have either `cpu` or `memory` defined

but the thing is that if `resources: ` without any properties will evaluate to `None` anyway so it's not strictly required in the current usage

will remove

Reply:
the same can be said with `min` and `max` as they all have `additionalProperties: false`

Reply:
fixed

Reply:
now you can get into a silly state with something like `resources: {}` I guess but that is not a biggie

Reply:
It's not that silly a state IMO.

Reply:
acceptable, I would say

Reply:
forcing strict structure can help with YAML funnies like

```
- endpoint: 
  resources:
  cpu:
    min: 0.1
    max: 1.0
```

but I think we always have `additionalProperties: false` and `resources: ` evaluates to `None` so it's not an issue